### PR TITLE
Turn on code style rules by default

### PR DESCRIPTION
**Pull request by davkean**
This turns on what I believe to be non-controversial code style rules on by default. Others which I found to be noisy, such as expression body and object initialization I've left turned off.

This a bit of an experiment over Christmas to see if it's too annoying, at which we can turn them from Error -> Info.

Note, I'll going to do one last pass to make sure we've fixed all current violations of these rules.

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/project-system @mavasani

File: build/Rulesets/Roslyn.ruleset [hunk 1]
```diff
@@ -1,115 +1,139 @@
...
+    <Rule Id="CA2101" Action="None" />
+    <Rule Id="RS0015" Action="Warning" />
+  </Rules>
+  <Rules AnalyzerId="System.Threading.Tasks.Analyzers" RuleNamespace="System.Threading.Tasks.Analyzers">
+    <Rule Id="RS0018" Action="Warning" />
+  </Rules>
+  <Rules AnalyzerId="XmlDocumentationComments.Analyzers" RuleNamespace="XmlDocumentationComments.Analyzers">
+    <Rule Id="RS0010" Action="Warning" />
+  </Rules>
+  <Rules AnalyzerId="Microsoft.CodeAnalysis.CSharp.Features" RuleNamespace="Microsoft.CodeAnalysis.CSharp.Features">
```
Comment (davkean) [inline build/Rulesets/Roslyn.ruleset]: This is where I've turned on the rules, not the AnalyzerId seems unneeded/ignored, as this also affects VB.

Comment (mavasani) [inline build/Rulesets/Roslyn.ruleset]: I would just remove CSharp from the AnalyzerId and RuleNamespace. Former is used when grouping rules in the ruleset editor. I believe the later is not used at all.

File: build/Rulesets/Roslyn.ruleset [hunk 2]
```diff
@@ -1,115 +1,139 @@
...
+    <Rule Id="IDE0003" Action="Error" />                      <!-- Remove qualification                         this.X = x;                                         X = x;                                -->
+    <Rule Id="IDE0005" Action="Error" />                      <!-- Using directive is unnecessary               using System.Text;                                                                        -->
+    <Rule Id="IDE0012" Action="Error" />                      <!-- Simplify framework names                     System.Int32 value;                                 int value;                            -->
+    <Rule Id="IDE0013" Action="Error" />                      <!-- Simplify framework names (member access)     Int32.Parse("1");                                   int.Parse("1");                       -->
+    <Rule Id="IDE0018" Action="Error" />                      <!-- Inline variable declaration                  TryParse(value, out result)                         TryParse(value, out int result)       -->
+    <Rule Id="IDE0018WithoutSuggestion" Action="Error" />
```
Comment (mavasani) [inline build/Rulesets/Roslyn.ruleset]: What are these rules with "WithoutSuggestion" suffix?

Comment (davkean) [inline build/Rulesets/Roslyn.ruleset]: No idea, I've asked the IDE - waiting on a response.

**Comment by davkean** [comment]
I've played around with - going to need to tweak, there's ~11000 errors in the AppDesigner/Editors code base.

**Pull request merged**